### PR TITLE
Uses RecordCursor in ReadNodeRecordsByCacheStep

### PR DESCRIPTION
**Pull request by tinwelint**
for more efficient reading of records.

### CODE REVIEW

File: community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/ReadNodeRecordsByCacheStep.java [hunk 1]
```diff
@@ -80,7 +95,9 @@ public void run()
         @Override
         public void change( long nodeId, ByteArray array )
         {
-            batch[cursor++] = nodeStore.getRecord( nodeId, nodeStore.newRecord(), RecordLoad.CHECK );
+            boolean inUse = recordCursor.next( nodeId );
+            assert inUse;
```
Comment (lutovich) [inline community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/ReadNodeRecordsByCacheStep.java]: Would it make sense to acquire cursor in `#start()` with `RecordLoad.NORMAL` and skip this assertion?

**Comment by lutovich** [comment]
@tinwelint lgtm, one small question

File: community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/ReadNodeRecordsByCacheStep.java (hunk 1 above)
Comment (tinwelint) [reply community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/ReadNodeRecordsByCacheStep.java]: Yeah, that's much better, I'll change it

**Comment by tinwelint** [comment]
@lutovich question resulted in a change for the better

**Pull request merged**